# Projet final

Dans cette étude, nous nous pencherons sur l'analyse des différentes caractéristiques des étoiles de notre galaxie. Les étoiles, objets célestes fascinants, possèdent des propriétés physiques variées telles que la température, la luminosité, la taille et la couleur, qui permettent de les classer en différentes catégories et types spectraux. À partir de l'échantillon de données disponible, nous analyserons comment ces caractéristiques interagissent et influencent la classification des étoiles, tout en explorant leur distribution et leur relation au sein de l'univers. Cette exploration vise à enrichir notre compréhension des étoiles et à offrir un aperçu détaillé de leurs propriétés essentielles.

Ces données ont été sélectionnées par Diaa ELdyin Essam Zaki, étudiante à la Faculté d'informatique et d'intelligence artificielle de l'Université du Caire. Nous les avons trouvées sur Kaggle.

In [2]:
# Import necessary libraries
import numpy as np
import pandas as pd
from scipy.stats import linregress

import plotly.express as px
import plotly.graph_objects as go

In [ ]:
stars = pd.read_csv ('https://github.com/YBIFoundation/Dataset/raw/main/Stars.csv')

In [10]:
print('Shape :', np.shape(stars))

print('Value counts :', '\n', stars['Star category'].value_counts())

Shape : (240, 8)
Value counts : 
 Star category
Brown Dwarf      40
Red Dwarf        40
White Dwarf      40
Main Sequence    40
Supergiant       40
Hypergiant       40
Name: count, dtype: int64


In [11]:
stars.head()

Temperature (K)  Luminosity (L/Lo)  Radius (R/Ro)  Absolute magnitude (Mv)  \
0             3068           0.002400         0.1700                    16.12   
1             3042           0.000500         0.1542                    16.60   
2             2600           0.000300         0.1020                    18.70   
3             2800           0.000200         0.1600                    16.65   
4             1939           0.000138         0.1030                    20.06   

   Star type Star category Star color Spectral Class  
0          0   Brown Dwarf        Red              M  
1          0   Brown Dwarf        Red              M  
2          0   Brown Dwarf        Red              M  
3          0   Brown Dwarf        Red              M  
4          0   Brown Dwarf        Red              M

**Quelques informations sur les unites dans notre dataframe :**

Le dataframe comporte une partie des étoiles de la galaxie (240, soit 40 étoiles de chaque type) -> une ligne = une étoile

On a donc bien 240 étoiles, choisies aléatoirement dans leur catégorie, et on a 8 colonnes. 

- Température absolue, en Kelvin
- Luminosité relative, en L/Lo où L = luminosité de l'étoile et Lo = luminosité du Soleil
- Rayon relatif, en R/RO où R = rayon de l'étoile et RO = rayon du Soleil
- Magnitude absoluen, en Mv
- Couleur de l'étoile
- Classe spectrale
- Type d'étoile

Pour info, Lo = 3.828 x 10^26 Watts et Ro = 6.9551 x 10^8 m

In [12]:
# Calcul de la régression linéaire avec scipy
x = stars['Temperature (K)']
y = stars['Luminosity (L/Lo)']
slope, intercept, r_value, p_value, std_err = linregress(x, y)

# Calcul des valeurs de la ligne de régression
x_line = np.linspace(min(x), max(x), 100)
y_line = slope * x_line + intercept

# Créer le graphique
fig = go.Figure()

# Ajouter le scatter plot des points de données
fig.add_trace(go.Scatter(
    x=x,
    y=y,
    mode='markers',
    name='Données',
    marker=dict(color='blue')
))

# Ajouter la ligne de régression avec l'équation et R^2
fig.add_trace(go.Scatter(
    x=x_line,
    y=y_line,
    mode='lines',
    name=f'Régression linéaire: y={slope:.2f}x + {intercept:.2f}<br>R²={r_value**2:.2f}',
    line=dict(color='green', width=2, dash='dot')
))

# Mise en forme du graphique
fig.update_layout(
    title='Relation entre la température et la luminosité des étoiles avec régression linéaire',
    xaxis_title='Température (K)',
    yaxis_title='Luminosité (L/Lo)',
    template='plotly_white',
    legend=dict(
        x=0.02,
        y=0.98,
        title_text='Légende'
    )
)

# Afficher le graphique
fig.show()

R^2 = 0.15 très éloigné de 1 donc pas du tout pertinent de faire une régression linéaire. Donc pas de corrélation entre la luminosité et la température.

In [ ]:
color_map = {
    'O': 'blue',
    'B': 'lightskyblue',
    'A': 'gray',
    'F': 'yellow',
    'G': 'gold',
    'K': 'orange',
    'M': 'red'
}

fig = px.scatter(stars, x='Temperature (K)', y='Luminosity (L/Lo)', color='Spectral Class',
                 color_discrete_map=color_map,
                 title='Luminosité des étoiles en fonction de leur température et classe spectrale',
                 labels={'Temperature (K)': 'Température (K)', 'Luminosity (L/Lo)': 'Luminosité (L/Lo)'})
fig.show()

In [13]:
# Dictionnaire des couleurs
color_map = {
    'Red': 'red',
    'Blue White': 'dodgerblue',
    'White': 'white',
    'Yellowish White': 'lightyellow',
    'Orange': 'orange',
    'Blue': 'blue',
    'Pale yellow orange': '#FFD580',
    'Blue-white': '#ADD8E6',
    'Whitish': '#F5F5F5',
    'yellow-white': '#FFFFE0',
    'White-Yellow': '#FFFACD'
}

# Graphique avec la palette de couleurs
fig = px.scatter(
    stars,
    x='Luminosity (L/Lo)',
    y=[1] * len(star),  # Une valeur constante pour l'axe y
    color='Star color',
    color_discrete_map=color_map,
    size='Luminosity (L/Lo)',
    labels={'y': 'Stars'},
    title='Couleur des étoiles en fonction de leur luminosité'
)

fig.show()

Apriori, on ne peut pas identifier de pattern clair entre la couleur et la luminosité des étoiles. En effet, nous verrons par la suite que la couleure de l'étoile est corrélée à la température, et non pas à la luminosité.

In [15]:
# Dictionnaire des couleurs
color_map = {
    'Red': 'red',
    'Blue': 'blue',
    'White': 'white',
    'Orange': 'orange',
    'Yellow': 'yellow',
    'Blue White': 'lightblue',
    'Blue-white': '#ADD8E6',
    'Orange-Red': '#FF4500',
    'White-Yellow': '#FFFACD',
    'Yellowish': '#FFFF00',
    'Pale yellow orange': '#FFD580',
    'Whitish': '#F5F5F5'
}

# Créer le graphique de dispersion avec les couleurs personnalisées
fig = px.scatter(
    stars,
    x='Temperature (K)',
    y=[1] * len(star),  # Valeur constante pour l'axe des ordonnées
    color='Star color',
    color_discrete_map=color_map,
    title='Couleur des étoiles en fonction de leur température',
    labels={'Temperature (K)': 'Température (K)', 'Star color': 'Couleur des étoiles'},
    template='plotly_white'
)

# Modifier l'apparence pour rendre le graphe plus intuitif
fig.update_yaxes(visible=False)  # Masquer l'axe des ordonnées

# Afficher le graphique
fig.show()

On observe que les étoiles de température plus basse tendent à être rouges ou oranges, correspondant aux étoiles plus froides. À mesure que la température augmente, les couleurs des étoiles se déplacent vers le bleu et le blanc, ce qui est caractéristique des étoiles plus chaudes. Cette transition confirme la relation bien établie entre la couleur et la température des étoiles, où des températures plus élevées entraînent un spectre plus bleu. Cela met en évidence les variations spectrales selon l'intensité de la chaleur des étoiles.

In [16]:
color_map = {
    'O': 'blue',
    'B': 'lightskyblue',
    'A': 'gray',
    'F': 'yellow',
    'G': 'gold',
    'K': 'orange',
    'M': 'red'
}

fig = px.scatter(stars, x='Absolute magnitude (Mv)', y='Radius (R/Ro)', color='Star category',
                 color_discrete_map=color_map,
                 title='Rayon des étoiles en fonction de leur magnitude et de leur catégorie')
fig.show()

Ainsi, les phases de la vie d'une étoile sont très liées à leur magnitude.